hpx/plugins/plugin_registry.hpp

```diff
@@ -57,11 +57,11 @@ namespace hpx { namespace plugins
         bool get_plugin_info(std::vector<std::string>& fillini)
         {
             using namespace boost::assign;
-            fillini += std::string("[hpx.plugins.") +
+            fillini += std::string("[" HPX_PLUGIN_CONFIG_SECTION ".plugins.") +
                 unique_plugin_name<plugin_registry>::call() + "]";
             fillini += "name = " HPX_PLUGIN_STRING;
-            fillini += std::string("path = ") +
-                util::find_prefixes("/hpx", HPX_PLUGIN_STRING);
+            fillini += std::string("path = ") + util::find_prefixes(
+                    "/" HPX_PLUGIN_PATH_SUFFIX, HPX_PLUGIN_STRING);
```

Comment:
Wouldn't it be better to have `HPX_PLUGIN_CONFIG_SECTION` and `HPX_PLUGIN_PATH_SUFFIX` as a parameter to the plugin registry instead? Making it relying on the global `config.hpp` header looks error prone. The following should work as well: https://wandbox.org/permlink/4emy0DIwSOzxmSjs

And provides a much cleaner interface for the module declarations.

Reply:
This will break NVCC most likely, but we might be able to keep this code out of view of this compiler.

Reply:
Done.